File: matura/liczbybin_z6.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cmath>

using namespace std;

string liner (int a){
	string l;
	ifstream plik;
	plik.open("liczby.txt");
	for (int i = 1; i<=a; i++) {
		plik >> l;
	}
	return l;
}

inline int digit(char a) {
    if ((int)a == 48) return 0;
    return 1;
}

int ptg (int l, int w) {
	if (w == 0)
		return 1;
	else
		return ptg(l, w-1) * l;
}

long int bindouble(string a, int lng){
  //int lng = a.length();
  long int collector = 0;
	int i = 0;
	while (lng > 0) {
		collector = collector + ptg(2,lng-1) * digit(a[i]);
		lng--;
		i++;
	}
	return collector;
}

int main(){
	int ilep = 0; //ile liczb parzystych
	long int lmax = 0; //liczba największa (10)
	string lmaxb=""; //liczba największa (2)
	long int l = 0; //liczba bieżąca (10)
	string lb=""; //liczba bieżąca (2)
	int lbdl=0; //długość liczby bieżącej (2)
	int ile9 = 0; //ile liczb 9-cyfrowych (2)
	long int suma9 = 0; //suma liczba 9 cyfrowych(10)
	for(int i = 0; i<1000; i++) {
		lb=liner(i+1);
		lbdl = lb.length();
		l = bindouble(lb,lbdl);
		if (l > lmax) {
			lmax = l;
			lmaxb = lb;
		}
		if (l % 2 == 0) ilep++;
		if (lbdl == 9) {
			ile9++;
			suma9 = suma9 + l;
		}
	}
	cout << "W pliku jest: " << ilep << " liczb parzystych" << endl;
	cout << "Największa liczba to: " << lmaxb << "(2), " << lmax << "(10)" << endl;
	cout << "Liczb 9-cyfrowych: " << ile9 << " Suma: " << suma9 << "(10)" << endl;
	return 0;
}

File: matura/liczbybin_z6.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

zrodlo = "liczby.txt"
lmax = 0
lmaxb =""

def bin2dec(liczba2, lbdl):
    liczba10 = 0;
    for i, lb in enumerate(liczba2):
        #print lb, 2**(dl-1-i) * ord()
        liczba10 = liczba10 + (2**(lbdl-1-i) * int(lb));

    return liczba10

lp = 0  # liczb parzystych
l9 = 0  # liczb 9-cyfrowych
suma9 = 0  # suma liczb 9-cyfrowych

with open(zrodlo) as f:
    for lb in f:
        lb = lb.replace("\n", "")
        lb = lb.replace("\r", "")
        lbdl = len(lb)  # długość liczby binarnej
        # l = int(lb, 2)  # wartość dziesiętna
        l = bin2dec(lb, lbdl)
        if (l > lmax):  # znajdź największą liczbę
            lmax = l
            lmaxb = lb
        if l % 2 == 0:  # zlicz liczby parzyste
            lp += 1
        if lbdl == 9:
            l9 += 1
            suma9 += l

print "Liczb parzystych:",lp
print "Największa:", lmaxb, lmax
print "9-cyfrowych:", l9
